src/category_theory/essentially_small.lean

```diff
@@ -0,0 +1,210 @@
+/-
+Copyright (c) 2021 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import logic.small
+import category_theory.skeletal
+
+/-!
+# Essentially small categories.
+
+A category given by `(C : Type u) [category.{v} C]` is `w`-essentially small
+if there exists a `small_model C : Type w` equipped with `[small_category (small_model C)]`.
+
+A category is `w`-locally small if every hom type is `w`-small.
+
+The main theorem here is that a category is `w`-essentially small iff
+the type `skeleton C` is `w`-small, and `C` is `w`-locally small.
+-/
+
+universes w v v' u u'
+
+open category_theory
+
+variables (C : Type u) [category.{v} C]
+
+namespace category_theory
+
+/-- A category is `essentially_small.{w}` if there exists
+an equivalence to some `S : Type w` with `[small_category w]`. -/
+class essentially_small (C : Type u) [category.{v} C] : Prop :=
+(equiv_small_category : ∃ (S : Type w) [small_category S], by exactI ∃ (e : C ≌ S), true)
```

Comment:
Is there an advantage to the pattern `Exists e, true` over `nonempty e`?

Reply:
Nope. :-) Just me apparently forgetting about `nonempty`...!? I'll fix.